src/create-user/controller.ts

```diff
@@ -27,6 +27,10 @@ class CreateUserController extends Controller {
     const nickname = ctx.request.body.nickname;
     const type = ctx.request.body.type;
 
+    if (type === 'user' && !principalService.isIdentityValid(identity, type)) {
+      throw new UnprocessableEntity('Identity must include mailto:');
+    }
+
     if (!identity || !identity.includes(':') || identity.includes(' ')) {
```

Comment:
This `if` statement seems to at least partially duplicate the new check you are doing. We should probably validate the identity just once.

Reply:
I do like the idea of having the check outside the controller and in the service, so it's reuable.